### PR TITLE
[ramda] fix R.flatten 

### PR DESCRIPTION
**Pull request by swwind**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#ordering
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

------

Old code matches a wrong type unexpectedly:

```js
R.flatten([[1, 2], [3, 4]]); // typeof number[][]
```

I think that use `R.flatten` to flatten a two-dimensional array is more general than one, thus I swapped the order of two overloaded functions.

Now it could match the type correctly as we need.

```js
R.flatten([[1, 2], [3, 4]]); // typeof number[]
R.flatten([1, 2, 3, 4]); // typeof number[]
```

### CODE REVIEW

File: types/ramda/ramda-tests.ts [hunk 1]
```diff
@@ -925,8 +925,8 @@ interface Obj {
 };
 
 () => {
-    R.flatten([1, 2, [3, 4], 5, [6, [7, 8, [9, [10, 11], 12]]]]);
-    // => [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    R.flatten([[1, 2], [2, 3], [3, 4]]); // => [1, 2, 3, 4, 5, 6]
```
Comment (Nemo108) [inline types/ramda/ramda-tests.ts]: wrong
`R.flatten([[1, 2], [2, 3], [3, 4]]);` will give you `[1, 2, 2, 3, 3, 4]`

File: types/ramda/index.d.ts [hunk 2]
```diff
@@ -1405,7 +1405,7 @@ declare namespace R {
          * Returns a new list by pulling every item out of it (and all its sub-arrays) and putting
          * them in a new array, depth-first.
          */
-        flatten<T>(x: ReadonlyArray<T> | ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>>): T[];
+        flatten<T>(x: ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>> | ReadonlyArray<T>): T[];
```
Comment (Nemo108) [inline types/ramda/index.d.ts]: what case does this swap solves?

Comment (swwind) [reply types/ramda/index.d.ts]: Maybe I need more tests?

Comment (Nemo108) [reply types/ramda/index.d.ts]: I've ran some. Current typings aren't the best, but it's not because of your changes.
Your contribution is good

**Review by Nemo108 — APPROVED** [review]

**Pull request merged**